src/java/htsjdk/samtools/reference/FastaSequenceFile.java

```diff
@@ -145,6 +145,9 @@ private String readSequenceName() {
                     bases = tmp;
             }
         }
+        // When length is known, make sure there is no trailing whitespace that hasn't been traversed.
+        skipToEoln();
```

Comment:
Why not put this in the if block on line 139 so it only applies to the 'length is known' case?  I am not sure it matters but I am being defensive.